swift/extractor/trap/TrapArena.h

```diff
@@ -0,0 +1,70 @@
+//
+// Created by redsun82 on 18.01.22.
+//
+
+#pragma once
+
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+#include "swift/extractor/trap/TrapLabel.h"
+
+namespace codeql {
+
+// TrapArena has the responsibilities to
+// * allocate distinct trap #-labels outputting their assignment to '*' or a @-keys to a trap output
+// * forwarding trap entries to said output
+// TODO: split or move these responsibilities into separate classes
+class TrapArena {
```

Comment:
I find the name very uninformative. Maybe just address the above todo, and split this into parts that do have informative names?

Reply:
yeah, I'm thinking about putting the printing responsibility in a `TrapFile` class, and moving the allocation to either the dispatcher, or the upcoming label store

Reply:
I've moved most of the `TrapArena` guts (the outputting part) to `TrapOutput`, also pulling in the trap temp file creation and moving into place.

With this, `TrapArena` is holding very little code, that will be moved elsewhere in upcoming PRs